src/mds/MDCache.cc

```diff
@@ -6542,13 +6548,17 @@ void MDCache::trim_lru(uint64_t count, expiremap& expiremap)
           << " pinned=" << lru.lru_get_num_pinned()
           << dendl;
 
-  for (;;) {
+  bool throttled = false;
+  while (1) {
+    throttled |= (size_t)trim_counter.get() >= trim_threshold;
+    if (throttled) break;
     CDentry *dn = static_cast<CDentry*>(bottom_lru.lru_expire());
     if (!dn)
       break;
     if (trim_dentry(dn, expiremap)) {
       unexpirables.push_back(dn);
     } else {
+      trim_counter.hit(1);
```

Comment:
this is expensive. we should do it in batch

Reply:
sure, fixed.